### PR TITLE
NE-1808: Bump k8s.io, openshift, controller-runtime modules and golang

### PR DESCRIPTION
**Pull request by alebedev87**
Bumped aws-load-balancer-controller to f39ae43121c3 to use latest CRD scheme in e2e tests.- Bumped golang to 1.22:
    - Update go.mod
    - Update Dockerfiles
- Bumped k8s.io/* modules to 0.30.3 and OpenShift API to 20240812094746-86145edb40cf.
- Bumped controller-runtime to 0.18.5:
    - Manager's `Port` option removed, now using dedicated webhook server field (https://github.com/kubernetes-sigs/controller-runtime/pull/2422).
    - Manager's `MetricsBindAddress` option  removed, now using dedicated metrics server field (https://github.com/kubernetes-sigs/controller-runtime/pull/2407).
    - Cache's `Namespaces` option replaced by `DefaultNamespaces` (https://github.com/kubernetes-sigs/controller-runtime/pull/2421).
- Bumped aws-load-balancer-controller to `f39ae43121c3` to use latest CRD scheme in e2e tests. 
- Regenerated CRD and bundle manifests using `make bundle` command.
- Bumped `kustomize` to v5 to fix a conflict caused by k8s.io bumps:
    - `kyaml` unable to use the bumped `github.com/google/gnostic-models/openapiv2` package.
- Removed `TypeMeta` from expected deployment object when it's compared to structured one retrieved from fake client (https://github.com/kubernetes-sigs/controller-runtime/pull/2633).
- Migrated infrastructure CRD retrieval to the new package `zz_generated.crd-manifests`:
    - Updated infrastructure CRD for unit tests using `make update-vendored-crds` command.
    
- Updated envtest setup to use the downstream index.
    - Infrastructure CRD uses CEL functions backported from newer k8s API.
    - Upstream envtest is not ready to use newer CEL function.
    - Bumped setup-envtest to be able to use --index flag.

Commands to reproduce the bump:
```
go get k8s.io/api@v0.30.3
go get k8s.io/client-go@v0.30.3
go get k8s.io/apiserver@v0.30.3
go get k8s.io/apiextensions-apiserver@v0.30.3
go get github.com/openshift/api@v0.0.0-20240812094746-86145edb40cf
go get sigs.k8s.io/controller-runtime@v0.18.5
go get sigs.k8s.io/kustomize/kustomize/v5@v5.4.3
o get sigs.k8s.io/aws-load-balancer-controller@f39ae43121c3f4de0129dda483c10b17a687491d
go get github.com/openshift/api/config/v1/zz_generated.crd-manifests@v0.0.0-20240812094746-86145edb40cf
go get sigs.k8s.io/controller-runtime/tools/setup-envtest@75a38d2123d4696d24d8d3c6b2966e7a34442653
go mod tidy
go mod vendor
make bundle
make update-vendored-crds
```

### CODE REVIEW

**Comment by alebedev87** [comment]
/test e2e-aws-proxy-operator

AWS jobs run slow recently (a rush before Christmas?).

**Comment by lihongan** [comment]
/retest

Step e2e-aws-proxy-operator-optional-operators-subscribe failed after 33m0s
FAIL: TestAWSLoadBalancerControllerWithWAFv2 (1020.22s)

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.19-openshift-4.12
+  tag: rhel-8-release-golang-1.22-openshift-4.17
```
Comment (lihongan) [inline .ci-operator.yaml:4]: can we change to `rhel-9-release` ?`

Comment (alebedev87) [reply .ci-operator.yaml:4]: RHEL9 migration is something we need to do, not only in this repository. Unfortunately I don't think I'll have enough time to do it for this release (before Christmas break). So, we need to file a bug for RHEL9 migration similar to [what we did for the externaldns](https://issues.redhat.com//browse/OCPBUGS-41683).

Comment (lihongan) [reply .ci-operator.yaml:4]: OK. then other look good.
@ShudiLi could you please help file a bug.

Comment (ShudiLi) [reply .ci-operator.yaml:4]: @lihongan Filed https://issues.redhat.com/browse/OCPBUGS-46442, thanks.

**Comment by alebedev87** [comment]
Bumped `aws-load-balancer-controller` module according to [the remark from PR#139](https://github.com/openshift/aws-load-balancer-operator/pull/139#issuecomment-2545950343).

**Comment by alebedev87** [comment]
`OutOfMemoryKilled`

/retest

**Comment by alebedev87** [comment]
Migrated infrastructure CRD retrieval to the new package `zz_generated.crd-manifests` after the merge of https://github.com/openshift/aws-load-balancer-operator/pull/137.

**Comment by alebedev87** [comment]
`OutOfMemoryKilled`

/retest

**Comment by ShudiLi** [comment]
tested it with 4.18.0-0.ci.test-2024-12-17-024008-ci-ln-gi4bdmb-latest 
1.
 % oc get clusterversion
NAME      VERSION                                                   AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.18.0-0.ci.test-2024-12-17-024008-ci-ln-gi4bdmb-latest   True        False         106m    Cluster version is 4.18.0-0.ci.test-2024-12-17-024008-ci-ln-gi4bdmb-latest

2.
% oc -n aws-load-balancer-operator  get pods                                                                    
NAME                                                              READY   STATUS      RESTARTS   AGE
792699d494754ea9f0ee4383a1ee7c9d515dd5720cf9b35fb9b2c31ecazbzcm   0/1     Completed   0          94m
aws-load-balancer-controller-cluster-5bb5f9fd9b-gk8bn             1/1     Running     0          49m
aws-load-balancer-operator-controller-manager-b88dc9679-tcb42     2/2     Running     0          94m
baeb04a3c4614c05dbbdbcc1844865253c12838d5adf8c2f896e79f14a1bbaf   1/1     Running     0          94m

3.
% go version -m manager 
manager: go1.22.7 (Red Hat 1.22.7-1.module+el8.10.0+22325+dc584f75)
	path	command-line-arguments
	dep	github.com/aws/aws-sdk-go-v2	v1.16.2	
	dep	github.com/aws/aws-sdk-go-v2/config	v1.13.1	
	dep	github.com/aws/aws-sdk-go-v2/credentials	v1.8.0	
	dep	github.com/aws/aws-sdk-go-v2/feature/ec2/imds	v1.10.0	
	dep	github.com/aws/aws-sdk-go-v2/internal/configsources	v1.1.9	
	dep	github.com/aws/aws-sdk-go-v2/internal/endpoints/v2	v2.4.3	
	dep	github.com/aws/aws-sdk-go-v2/internal/ini	v1.3.5	
	dep	github.com/aws/aws-sdk-go-v2/service/ec2	v1.29.0	
	dep	github.com/aws/aws-sdk-go-v2/service/internal/presigned-url	v1.7.0	
	dep	github.com/aws/aws-sdk-go-v2/service/sso	v1.9.0	
	dep	github.com/aws/aws-sdk-go-v2/service/sts	v1.14.0	
	dep	github.com/aws/smithy-go	v1.11.2	
	dep	github.com/beorn7/perks	v1.0.1	
	dep	github.com/cespare/xxhash/v2	v2.2.0	
	dep	github.com/davecgh/go-spew	v1.1.2-0.20180830191138-d8f796af33cc	
	dep	github.com/emicklei/go-restful/v3	v3.11.0	
	dep	github.com/evanphx/json-patch/v5	v5.9.0	
	dep	github.com/fsnotify/fsnotify	v1.7.0	
	dep	github.com/go-logr/logr	v1.4.1	
	dep	github.com/go-logr/zapr	v1.3.0	
	dep	github.com/go-openapi/jsonpointer	v0.19.6	
	dep	github.com/go-openapi/jsonreference	v0.20.2	
	dep	github.com/go-openapi/swag	v0.22.4	
	dep	github.com/gogo/protobuf	v1.3.2	
	dep	github.com/golang/groupcache	v0.0.0-20210331224755-41bb18bfe9da	
	dep	github.com/golang/protobuf	v1.5.4	
	dep	github.com/google/gnostic-models	v0.6.8	
	dep	github.com/google/go-cmp	v0.6.0	
	dep	github.com/google/gofuzz	v1.2.0	
	dep	github.com/google/uuid	v1.3.1	
	dep	github.com/imdario/mergo	v0.3.13	
	dep	github.com/jmespath/go-jmespath	v0.4.0	
	dep	github.com/josharian/intern	v1.0.0	
	dep	github.com/json-iterator/go	v1.1.12	
	dep	github.com/mailru/easyjson	v0.7.7	
	dep	github.com/matttproud/golang_protobuf_extensions	v1.0.4	
	dep	github.com/modern-go/concurrent	v0.0.0-20180306012644-bacd9c7ef1dd	
	dep	github.com/modern-go/reflect2	v1.0.2	
	dep	github.com/munnerz/goautoneg	v0.0.0-20191010083416-a7dc8b61c822	
	dep	github.com/openshift/api	v0.0.0-20240812094746-86145edb40cf	
	dep	github.com/openshift/aws-load-balancer-operator	(devel)	
	dep	github.com/openshift/cloud-credential-operator	v0.0.0-20230816031419-2c3298b1bb3a	
	dep	github.com/operator-framework/operator-lib	v0.11.0	
	dep	github.com/pkg/errors	v0.9.1	
	dep	github.com/prometheus/client_golang	v1.16.0	
	dep	github.com/prometheus/client_model	v0.4.0	
	dep	github.com/prometheus/common	v0.44.0	
	dep	github.com/prometheus/procfs	v0.12.0	
	dep	github.com/spf13/pflag	v1.0.5	
	dep	go.uber.org/multierr	v1.11.0	
	dep	go.uber.org/zap	v1.26.0	
	dep	golang.org/x/exp	v0.0.0-20220722155223-a9213eeb770e	
	dep	golang.org/x/net	v0.23.0	
	dep	golang.org/x/oauth2	v0.12.0	
	dep	golang.org/x/sys	v0.18.0	
	dep	golang.org/x/term	v0.18.0	
	dep	golang.org/x/text	v0.14.0	
	dep	golang.org/x/time	v0.3.0	
	dep	gomodules.xyz/jsonpatch/v2	v2.4.0	
	dep	google.golang.org/protobuf	v1.33.0	
	dep	gopkg.in/inf.v0	v0.9.1	
	dep	gopkg.in/yaml.v2	v2.4.0	
	dep	gopkg.in/yaml.v3	v3.0.1	
	dep	k8s.io/api	v0.30.3	
	dep	k8s.io/apiextensions-apiserver	v0.30.3	
	dep	k8s.io/apimachinery	v0.30.3	
	dep	k8s.io/client-go	v0.30.3	
	dep	k8s.io/klog/v2	v2.120.1	
	dep	k8s.io/kube-openapi	v0.0.0-20240228011516-70dd3763d340	
	dep	k8s.io/utils	v0.0.0-20240502163921-fe8a2dddb1d0	
	dep	sigs.k8s.io/controller-runtime	v0.18.5	
	dep	sigs.k8s.io/json	v0.0.0-20221116044647-bc3834ca7abd	
	dep	sigs.k8s.io/structured-merge-diff/v4	v4.4.1	
	dep	sigs.k8s.io/yaml	v1.4.0	
	build	-buildmode=exe
	build	-compiler=gc
	build	-tags=strictfipsruntime
	build	CGO_ENABLED=1
	build	CGO_CFLAGS=
	build	CGO_CPPFLAGS=
	build	CGO_CXXFLAGS=
	build	CGO_LDFLAGS=
	build	GOARCH=amd64
	build	GOOS=linux
	build	GOAMD64=v1

4.
%oc get crd  infrastructures.config.openshift.io -oyaml

5. created the ingress and then curl the ALB with the specified host

**Comment by ShudiLi** [comment]
/label qe-approved
thanks

**Comment by alebedev87** [comment]
To fix the unit test:
- Updated envtest setup to use the downstream index.
    - Infrastructure CRD uses CEL functions backported from newer k8s API.
    - Upstream `envtest` is not ready to use newer CEL function.
    - Bumped `setup-envtest` to be able to use `--index` flag.

**Comment by alebedev87** [comment]
/label px-approved

The docs are enough for this feature.

**Comment by ShudiLi** [comment]
/label qe-approved
thanks

**Pull request merged**